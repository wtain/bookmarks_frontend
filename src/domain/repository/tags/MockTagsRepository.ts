import ReactHelpers from '../../../utils/ReactHelpers';
import ITagsRepository from './ITagsRepository';


class MockTagsRepository implements ITagsRepository {
  async getTags(): Promise<string[]> {
    await ReactHelpers.delay(300);
    return ["tag1", "tag2", "tag3"];
  }
  
}

export default MockTagsRepository;