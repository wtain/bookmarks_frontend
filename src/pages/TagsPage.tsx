
import React, { useEffect, useState } from "react"
import { useNavigate } from "react-router-dom";
import Tag from "../components/UI/TagsEditor/Tag";
import ITagsRepository from "../domain/repository/tags/ITagsRepository";

interface Props {
    tagsRepository: ITagsRepository;
}

const TagsPage = (props: Props) => {

    const navigate = useNavigate();

    let [tags, setTags] = useState(
        [] as string[]
    )

    let [loading, setLoading] = useState(false);

    const loadTags = async(success: (tags: string[]) => void, error: (e: any) => void) => {
        try {
            const tags = await props.tagsRepository.getTags();
            success(tags)
        } catch (e) {
            error(e)
        }
    }

    useEffect(() => {
        setLoading(true);
        loadTags((tags: string[]) => {
            setTags(tags)
            setLoading(false);
        }, 
            (e) => {
                setLoading(false);
            })
    }, [])

    return (
        <>
            {tags.map(tag => <Tag key={tag} tag={{name: tag, id: ""}} onClick={() => {
                navigate("/bookmarks/" + tag)
            }} />)}
        </>
    )
}

export default TagsPage;