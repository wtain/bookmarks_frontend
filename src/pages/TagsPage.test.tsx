import renderer from 'react-test-renderer';
import MockTagsRepository from '../domain/repository/tags/MockTagsRepository';
import { BrowserRouter } from 'react-router-dom';
import TagsPage from './TagsPage';

it('TagsPage renders correctly', () => {
  const tagsRepository = new MockTagsRepository();

  const tree = renderer
    .create(
      <BrowserRouter>
        <TagsPage tagsRepository={tagsRepository} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});