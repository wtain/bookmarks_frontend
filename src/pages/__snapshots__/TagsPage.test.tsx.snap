// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsPage correctly 1`] = `
<span
  className="tag"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  (Clear)
</span>
`;

exports[`TagsPage renders correctly 1`] = `
<span
  className="tag"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  (Clear)
</span>
`;
